org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DerivedResourcesUpdater.java

```diff
@@ -139,4 +157,37 @@ static void update(IProject workspaceProject, OmniEclipseProject project, Persis
         new DerivedResourcesUpdater(workspaceProject, project).update(persistentModel, monitor);
     }
 
+    /**
+     * Helper class to collect derived resources.
+     */
+    private static final class DerivedResourcesInfo {
```

Comment:
This is not just about derived resources, but "special folders", probably should rename this and the containing class.